provision/kubernetes/deploy.go

```diff
@@ -590,6 +590,19 @@ func createAppDeployment(client *ClusterClient, oldDeployment *v1beta2.Deploymen
 	_, tag := image.SplitImageName(imageName)
 	expandedLabels["version"] = tag
 	expandedLabelsNoReplicas["version"] = tag
+	containerPorts := make([]apiv1.ContainerPort, len(processPorts))
+	for i, port := range processPorts {
+		var portInt int
+		if port.TargetPort > 0 {
+			portInt = port.TargetPort
+		} else if port.Port > 0 {
+			portInt = port.Port
+		} else {
+			portInt, _ = strconv.Atoi(provision.WebProcessDefaultPort())
+		}
+		containerPorts[i].ContainerPort = int32(portInt)
```

Comment:
I don't think it's worth. A duplicated port would lead to an error creating the deployment, and the error message clearly shows what's wrong, like in this example:

```
Service "multiport-web" is invalid: spec.ports[2]: Duplicate value: core.ServicePort{Name:"", Protocol:"TCP", Port:9999, TargetPort:intstr.IntOrString{Type:0, IntVal:0, StrVal:""}, NodePort:0}
```

Reply:
Imagine we have a the following configuraration: 

```
"kubernetes": {
    "groups": {
        "pod1": {
            "web": {
                "ports": [
                    {
                        "port": 1080,
                        "target_port": 8000
                    },
                    {
                        "port": 2080,
                        "target_port": 8000
                    }
                ]
            }
        }
    }
}
```

It is valid, but it would fail here because we would have 2 entries setting `ContainerPort` to `8000`. In this situation having a single port entry would be the right thing to do.